object-cache.php

```diff
@@ -1264,6 +1264,8 @@ public function build_client_parameters( $redis_server ) {
 		if ( file_exists( $redis_server['host'] ) && 'socket' === filetype( $redis_server['host'] ) ) { // unix socket connection.
 			// port must be null or socket won't connect.
 			$port = null;
+		} else { // tcp connection.
+			$port = ! empty( $redis_server['port'] ) ? $redis_server['port'] : 6379;
```

Comment:
$port already has a default value of 6379 at the top of this function.

```suggestion
		} else if( ! empty( $redis_server['port'] ) ) { // tcp connection.
			$port = $redis_server['port'];
```

Reply:
IMO 
```
$redis_server = [
'host' => '127.0.0.1',
'port' => $port,
'database' => $database,
];
``` 
should just be set at the top of the func instead of $port and $database and that clears out some `else` mess above, but I've not 👀 the implication of that default hostname yet.

Reply:
The suggested code change is not functionally different than the existing code -- it just lacks a ternary. 

```php
} elseif ( ! empty( $redis_server['port'] ) ) {
    $port = $redis_server['port'];
}
```

is functionally no different than:

```php
} else {
    $port = ! empty( $redis_server['port'] ? $redis_server['port'] : 6379;
}
```
...the only exception being that last `6379` could probably be changed to `$port` instead.

Reply:
It's functionally no different, but having `else {<another if>}` is messy and we should be relying on the earlier default value rather than having a second default defined in the ternary.

`$port` is _already_ 6379  as of the top of the function.

Reply:
Okay....but isn't `elseif` just `else (<another if>)`? Semantics, I know. I see what you're getting at.